plugin/pkg/scheduler/algorithm/priorities/util/topologies.go

```diff
@@ -76,3 +77,12 @@ func (tps *Topologies) NodesHaveSameTopologyKey(nodeA, nodeB *api.Node, topology
 		return nodesHaveSameTopologyKeyInternal(nodeA, nodeB, topologyKey)
 	}
 }
+
+// NodesHaveSameTopologyKey checks if nodeA and nodeB have same label value with given topologyKey as label key.
+// Empty topologyKey is not allowed.
+func NodesHaveSameTopologyKey(nodeA, nodeB *api.Node, topologyKey string) (bool, error) {
+	if len(topologyKey) == 0 {
```

Comment:
Coupling this utility function to the semantics of `PreferredDuringScheduling` is a little surprising. Could you do this validation at the (2) call-sites instead? That would also allow you to just rename/export `nodesHaveSameTopologyKeyInternal` and reduce the logic duplication going on here.